### ISSUE TITLE
Deprecate --service-* CLI args and replace with bootstrap config/overrides

### ISSUE THREAD

**Issue by mattklein123** [envoyproxy/envoy#3326]
All locality information cannot be set with the old CLI arguments and this causes confusion.

**Comment by derekargueta** [comment]
Interested in picking this up. I suppose this involves adding a deprecation notice when running with the `--service-*` CLI args, followed with removal of these args?

**Comment by mattklein123** [comment]
@derekargueta yup

**Comment by derekargueta** [comment]
@mattklein123 For this I imagine we still want to increment `deprecated_feature_use` for each deprecated flag used? (I ask because if so then a bit of extra plumbing will be needed since option parsing happens before the runtime is created)

**Comment by mattklein123** [comment]
Yeah for anything we want to deprecate at this point we have to both warn, increment the stat, and go through the multi-version deprecation with warn -> fail -> delete.
